source/extensions/tracers/zipkin/span_context_extractor.cc

```diff
@@ -28,13 +28,16 @@ bool getSamplingFlags(char c, const Tracing::Decision tracing_decision) {
 
 } // namespace
 
-SpanContextExtractor::SpanContextExtractor(Tracing::TraceContext& trace_context)
-    : trace_context_(trace_context) {}
+SpanContextExtractor::SpanContextExtractor(Tracing::TraceContext& trace_context,
+                                           bool w3c_fallback_enabled)
+    : trace_context_(trace_context), w3c_fallback_enabled_(w3c_fallback_enabled) {}
 
 SpanContextExtractor::~SpanContextExtractor() = default;
 
 bool SpanContextExtractor::extractSampled(const Tracing::Decision tracing_decision) {
   bool sampled(false);
+
+  // Try B3 single format first
```

Comment:
style-nit: comments start with capital letter and end with '.' (here and in other places)